apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -267,24 +302,44 @@ const aichatSlice = createSlice({
   initialState,
   reducers: {
     addChatMessage: (state, action: PayloadAction<ChatCompletionMessage>) => {
-      const newMessageId =
-        state.chatMessages[state.chatMessages.length - 1].id + 1;
+      const newMessageId = state.chatMessages.length
+        ? state.chatMessages[state.chatMessages.length - 1].id + 1
+        : 1;
       const newMessage = {
         ...action.payload,
         id: newMessageId,
       };
       state.chatMessages.push(newMessage);
     },
-    removeChatMessage: (state, action: PayloadAction<number>) => {
-      const updatedMessages = state.chatMessages.filter(
-        message => message.id !== action.payload
+    removeModelUpdateMessage: (state, action: PayloadAction<number>) => {
```

Comment:
Tried to make this reducer name specific (and only work if the message type is `MODEL_UPDATE` so that it is less likely to be accidentally used by a future developer to delete a single actual chat message.